components/style/values/specified/basic_shape.rs

```diff
@@ -770,8 +770,8 @@ impl Parse for BorderRadius {
 
 fn parse_one_set_of_border_values(context: &ParserContext, mut input: &mut Parser)
                                  -> Result<[LengthOrPercentage; 4], ()> {
-    let a = try!(LengthOrPercentage::parse(context, input));
 
+    let a = try!(LengthOrPercentage::parse_non_negative(input));
```

Comment:
The `context` becoming unused is fine, but what do you mean by, "other didn't seem to make a difference"? Are you saying that `border-radius: 2px -5px` didn't show a difference before and after you made your changes? After your changes, this should serialize to `border-radius: 2px`.